### PR TITLE
[Bug] Preventing Duplicate Species in Trainer Battles

### PR DESCRIPTION
**Pull request by frutescens**
## What are the changes the user will see?
Users will no longer have to fight trainers with repeated Pokemon. The game will try to supply unique party members instead. 

## Why am I making these changes?
Quid-pro-quo

## Screenshots
Can't prove definitely that no repeats will occur, but ran through Wave 112 multiple times. 
Console Output: 
image
Enemy Parties (Wave 112):
image
image
image
image
image

## What are the changes from a developer perspective?
A new function was created 'checkDuplicateSpecies' that checks if the Pokemon Species passed is already found in the party. If it is, it prompts the genNewPartyMemberSpecies to reroll again. An infinite loop is already prevented in the code as the game allows up to 10 retries before settling with the Pokemon chosen. 

## How to test the changes?
Choose any trainer that lacks fixed party members. Would recommend Wave 112.
Evil team leaders and the rivals are not a good choice b/c the all/majority of their party is set in stone.  

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/field/trainer.ts [hunk 1]
```diff
@@ -425,6 +425,12 @@ export default class Trainer extends Phaser.GameObjects.Container {
...
 
+    // Prompts reroll of party member species if species already present in the enemy party
+    if (this.checkDuplicateSpecies(ret)) {
+      console.log("Duplicate species detected, prompting reroll...");
+      retry = true;
+    }
+
     if (retry && (attempt || 0) < 10) {
       console.log("Rerolling party member...");
       ret = this.genNewPartyMemberSpecies(level, strength, (attempt || 0) + 1);
```
Comment (DayKev) [inline src/field/trainer.ts:436]:
```suggestion
    if (retry && (attempt ?? 0) < 10) {
      console.log("Rerolling party member...");
      ret = this.genNewPartyMemberSpecies(level, strength, (attempt ?? 0) + 1);
```
Might as well fix nearby.

**Comment by frutescens** [comment]
Seems to work with Wave 114 (Evil Team admins too)
image

**Review by DayKev — APPROVED** [review]

**Review by innerthunder — APPROVED** [review]
I think I'd rather have the pool filtered to unselected species instead of forcing rerolls in the future, but this seems like an ok temporary fix for now.

**Comment by xsn34kzx** [comment]
+1 to @innerthunder's comment, I feel like the root cause is the pool not "shrinking" as species get picked during team generation (this probably falls under a trainer generation rewrite anyway). In any case, lgtm!

**Comment by CodeTappert** [comment]
image

Seems fine. I will test some more

**Review by xsn34kzx — APPROVED** [review]

**Review by CodeTappert — APPROVED** [review]

**Pull request merged**